eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +115,101 @@ function GetPackageLookup($packageList) {
   return $packageLookup
 }
 
+function create-metadata-table($absolutePath, $readmeName, $moniker, $msService, $clientTableLink, $mgmtTableLink, $serviceName)
+{
+  $readmePath = Join-Path $absolutePath -ChildPath $readmeName
+  New-Item -Path $readmePath -Force
+  $lang = $LanguageDisplayName
+  $langTitle = "Azure $serviceName SDK for $lang"
+  $header = GenerateDocsMsMetadata -lang $lang -langTitle $langTitle -serviceName $serviceName `
+    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+    -msService $msService
+  Add-Content -Path $readmePath -Value $header
+
+  # Add tables, seperate client and mgmt.
+  $readmeHeader = "# $langTitle - $moniker"
+  Add-Content -Path $readmePath -Value $readmeHeader
+  if (Test-Path (Join-Path $absolutePath -ChildPath $clientTableLink)) {
+    $clientTable = "## Client packages - $moniker`r`n"
+    $clientTable += "[!INCLUDE [client-packages]($clientTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $clientTable
+  }
+  if (Test-Path (Join-Path $absolutePath -ChildPath $mgmtTableLink)) {
+    $mgmtTable = "## Management packages - $moniker`r`n"
+    $mgmtTable += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $mgmtTable -NoNewline
+  }
+}
+
+# Update the metadata table on attributes: author, ms.author, ms.service
+function update-metadata-table($absolutePath, $readmePath, $serviceName, $msService)
+{
+  $readmeContent = Get-Content -Path (Join-Path $absolutePath -ChildPath $readmeName) -Raw
+  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
+  $restContent = $Matches["content"]
+
+  $lang = $LanguageDisplayName
+  $metadataString = GenerateDocsMsMetadata -lang $lang -serviceName $serviceName `
+    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+    -msService $msService
+  Set-Content -Path $readmePath -Value "---`n$metadataString---`n$restContent" -NoNewline
```

Comment:
Isn't the `---` header markers already included in the metadataString now since you are calling GenerateDocsMsMetadata.